flake.nix

```diff
@@ -27,8 +27,84 @@
                         base
                         wai
                         text
+
+                        # Uncomment on local development for testing
+                        # hspec
                     ];
                 };
+
+                # Custom configuration that will start with `devenv up`
+                devenv.shells.default = {
+                    # Start Mailhog on local development to catch outgoing emails
+                    services.mailhog.enable = true;
+
+                    # Custom processes that don't appear in https://devenv.sh/reference/options/
+                    processes = {
+                        # Uncomment if you use tailwindcss.
+                        # tailwind.exec = "tailwindcss -c tailwind/tailwind.config.js -i ./tailwind/app.css -o static/app.css --watch=always";
+                    };
+                };
+            };
+
+            # Adding the new NixOS configuration for "qa"
+            # Used to deploy the IHP application to AWS.
+            #
+            # Change the `CHANGE-ME` to your correct config.
+            flake.nixosConfigurations."qa" = nixpkgs.lib.nixosSystem {
+                system = "x86_64-linux";
+                specialArgs = inputs;
+                modules = [
+                    "${nixpkgs}/nixos/modules/virtualisation/amazon-image.nix"
+                    ihp.nixosModules.appWithPostgres
+                    ({ lib, pkgs, ... }: {
+
+                        networking.firewall = {
+                            enable = true;
+                            allowedTCPPorts = [ 22 80 443 8000 ];
+                        };
+
+                        # Enable the Let's encrypt certificate
+                        security.acme.defaults.email = "CHANGE-ME@example.com";
+
+                        # Accept the terms of service of the Let's encrypt provider.
+                        security.acme.acceptTerms = true;
+
+                        services.nginx = {
+                            virtualHosts."CHANGE-ME.com" =  {
+                                # Uncomment to have http auth with username `foo` and password `bar`.
+                                # basicAuth = { foo = "bar"; };
+                            };
+                        };
+
+                        services.ihp = {
+                            domain = "CHANGE-ME.com";
+                            migrations = ./Application/Migration;
+                            schema = ./Application/Schema.sql;
+                            fixtures = ./Application/Fixtures.sql;
+                            sessionSecret = "CHANGE-ME";
+                            additionalEnvVars = {
+                                # Indicate the environment name, e.g. "production", "staging", "qa".
+                                ENV_NAME = "qa";
```

Comment:
Maybe we can/should auto-inject the app name (`qa` from the `flake.nixosConfigurations."qa"` name) instead?

Reply:
why is ENV_NAME needed at all?

Reply:
This is what I have in mind eventually - https://www.drupal.org/project/environment_indicator

(Changing some color property on the header, based on ENV, so you don't click the wrong things on LIVE, because you though it was local or DEV)

Reply:
Can't we use `IHP_ENV` for that? https://github.com/digitallyinduced/ihp/blob/master/NixSupport/nixosModules/options.nix#L56

Reply:
Yeah, I'll remove it.